pyro/contrib/gp/models/svgp.py

```diff
@@ -78,25 +88,18 @@ def guide(self):
         Xu = self.get_param("Xu")
 
         # define variational parameters
-        if self.y.dim() == 1:
-            mu_0 = torch.tensor(Xu.data.new(Xu.size(0)).zero_(),
-                                requires_grad=True)
-        else:
-            mu_0 = torch.tensor(Xu.data.new(Xu.size(0), self.y.size(1)).zero_(),
-                                requires_grad=True)
+        mu_shape = self.latent_shape + Xu.size()[:1]
+        mu_0 = torch.tensor(Xu.new(mu_shape).zero_(), requires_grad=True)
         mu = pyro.param("u_loc", mu_0)
-        unconstrained_Lu_0 = torch.tensor(Xu.new(Xu.size(0), Xu.size(0)).zero_(),
-                                          requires_grad=True)
+        Lu_shape = self.latent_shape + torch.Size([Xu.size(0), Xu.size(0)])
+        # TODO: use new syntax for pyro.param constraint
+        unconstrained_Lu_0 = torch.tensor(Xu.new(Lu_shape).zero_(), requires_grad=True)
         unconstrained_Lu = pyro.param("unconstrained_u_tril", unconstrained_Lu_0)
-        Lu = transform_to(constraints.lower_cholesky)(unconstrained_Lu)
+        Lu = batch_lower_cholesky_transform(unconstrained_Lu)
 
-        # TODO: use scale_tril=Lu
-        Ku = Lu.t().matmul(Lu) + self.jitter.expand(Lu.size(0)).diag()
-        # correct event_shape for mu
-        mu_t = mu.t() if mu.dim() == 2 else mu
-        pyro.sample("u", dist.MultivariateNormal(loc=mu_t, covariance_matrix=Ku).reshape(
-            extra_event_dims=mu_t.dim() - 1))
-        return kernel, likelihood, Xu, mu, Lu
+        pyro.sample("u", dist.MultivariateNormal(loc=mu, scale_tril=Lu)
+                    .reshape(extra_event_dims=mu.dim()-1))
```

Comment:
Is `mu` batched or is it multidimensional? If it is multidimensional then you've correctly used `extra_event_dims`, but if its batched, then it may be more appropriate to use `iarange`.

Reply:
I think mu batched: `u_shape = D x N`, D is batch_shape, N is event_shape. Is there any difference if we use `iarange` here instead of `extra_event_dims`? Here D can be arbitrary shape: `torch.Size([])` or `torch.Size([3, 9, 7])`.